File: fib.py
#! /usr/bin/env python
import sys
import time

def fib(n):
	if (n >= 3):
		return fib(n-1) + fib(n-2)
	else:
		return 1

if len(sys.argv) != 2:
	print("Usage: python fib.py <n>")
	exit()

for i in range(1, int(sys.argv[1]) + 1):
	tpstart = time.time()
	result = fib(i)
	tpend = time.time()
	timeuse = tpend - tpstart
	print("Calculated Fib(%s) = %s, Time Used: %ss" % (i, result, timeuse))

File: fib.rb
#!/usr/bin/ruby
def fib(n)
	if (n >= 3)
		return fib(n-1) + fib(n-2)
	else
		return 1
	end
end

if ARGV.length != 1
	puts("Usage: ruby fib.rb <n>")
	exit()
end

for i in 1..ARGV.first.to_i()
	tpstart = Time.now()
	result = fib(i)
	tpend = Time.now()
	timeuse = tpend - tpstart
	puts("Calculated Fib(#{i}) = #{result}, Time Used: #{timeuse}s")
end
